homeassistant/components/auth/__init__.py

```diff
@@ -637,8 +643,9 @@ async def websocket_delete_all_refresh_tokens(
     else:
         connection.send_result(msg["id"], {})
 
-    # This will close the connection so we need to send the result first.
-    hass.loop.call_soon(hass.auth.async_remove_refresh_token, current_refresh_token)
+    if not limit_token_types or current_refresh_token.token_type == token_type:
```

Comment:
I added the same guard as above to ensure we don't delete the current token unless it matches the time if there is a limit on which tokens to delete